src/util/prolog/prolog.test.ts

```diff
@@ -1,296 +1,128 @@
 
 import { PrologSession } from "./prologSession";
-import * as program from "./testPrograms";
+import * as programs from "./testPrograms";
 
-// Expected results are consistent with SWI-Prolog behavior except that they usually have an extra "false."
+// Tau-Prolog always uses "false." to signify the end of the output stream. This behavior deviates from
+// SWI Prolog's behavior sometimes, but it generally more consistent.
 
-describe("Tau-Prolog Behavior, PrologSession.executeQuery() & PrologResult.getResultArray()", () => {
-	describe("Simple Programs", () => {
-		test("Truth Value (True)", async () => {
-				const session = new PrologSession("likes(lisa, bob).", 1000);
-				const results = (await session.executeQuery("likes(lisa, bob).")).getResultArray();
+describe("Tau-Prolog Behavior", () => {
+		test("Evaluates Facts Correctly (True Queries)", async () => {
+			const session = programs.evenNumbersFacts;
+			const results = (await session.executeQuery("even(2).")).getRawResults();
 
-				expect(results).toStrictEqual([["true"], ["false"]]);
+			expect(results).toStrictEqual(["true ;", "false."]);
 		});
 
-		test("Truth Value (False)", async () => {
-				const session = new PrologSession("likes(lisa, bob).", 1000);
-				const results = (await session.executeQuery("likes(lisa, dieter).")).getResultArray();
+		test("Evaluates Facts Correctly (False Queries)", async () => {
+				const session = programs.evenNumbersFacts;
+				const results = (await session.executeQuery("even(3).")).getRawResults();
 
-				expect(results).toStrictEqual([["false"]]);
+				expect(results).toStrictEqual(["false."]);
 		});
 
-		test("One Variable One Result", async () => {
-
-				const session = new PrologSession("likes(lisa, bob).", 1000);
-				const results = (await session.executeQuery("likes(lisa, X).")).getResultArray();
-				expect(results).toStrictEqual([["bob"], ["false"]]);
+		test("Evaluates Equalities Correctly (True Queries)", async () => {
+				const session = programs.emptyCode;
+				const results = (await session.executeQuery("2 = 2.")).getRawResults();
 
+				expect(results).toStrictEqual(["true ;", "false."]);
```

Comment:
I wonder: Does prolog always use `;` to separate individual results and `.` at the very end? In that case, they don't offer any more information in the output and we could omit them (returning/expecting `["true", "false"]` instead).

Reply:
But then again, this method is called `rawResults`. So it's probably fine like that.